### DISCUSSION TITLE
[abctl] failed to verify certificate: x509

### DISCUSSION THREAD

**Discussion by atlasrule** [airbytehq/airbyte#45087] [Deployment] [answered]
### What happened?

Hello,

When I start abctl.exe on windows host using wsl docker, I get the following unkown authority certificate error:
 
  INFO    Using Kubernetes provider:
            Provider: kind
            Kubeconfig: C:\Users\atlasrule\.airbyte\abctl\abctl.kubeconfig
            Context: kind-airbyte-abctl
 SUCCESS  Found Docker installation: version 27.0.3
 SUCCESS  Port 8000 appears to be available
 SUCCESS  Existing cluster 'airbyte-abctl' found
 SUCCESS  Cluster 'airbyte-abctl' validation complete
  ERROR   Failed to initialize 'local' command
  ERROR   unable to initialize local command: error communicating with kubernetes: unable to fetch kubernetes server version: **Get "https://127.0.0.1:61690/version": tls: failed to verify certificate: x509: certificate signed by unknown authority**

  INFO    An error occurred while communicating with the Kubernetes cluster.
          If this error persists, you may need to run the uninstall command before attempting to run
          the install command again.

![ssl-error](https://github.com/user-attachments/assets/511bf21c-e109-4e14-87c6-ceadc6d9dc16)

### What did you expect to happen?

Couldn't find any bypass or ignore certificate flag in abctl, and also couldn't edit the dockerfile for installing any certificate, can you help me about the convention for this error?

### Abctl Version

<details>

```console
$ abctl version
version: v0.12.2
```

</details>

### Docker Version

<details>

```console
$ docker version
Client:
 Version:           27.0.3
 API version:       1.46
 Go version:        go1.21.11
 Git commit:        7d4bcd8
 Built:             Sat Jun 29 00:03:32 2024
 OS/Arch:           windows/amd64
 Context:           desktop-linux

Server: Docker Desktop 4.32.0 (157355)
 Engine:
  Version:          27.0.3
  API version:      1.46 (minimum version 1.24)
  Go version:       go1.21.11
  Git commit:       662f78c
  Built:            Sat Jun 29 00:02:50 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.7.18
  GitCommit:        ae71819c4f5e67bb4d5ae76a6b735f29cc25774e
 runc:
  Version:          1.7.18
  GitCommit:        v1.1.13-0-g58aa920
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

</details>

### OS Version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here

# On Mac:
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
BuildNumber  Caption                   OSArchitecture  Version                                                          19045        Microsoft Windows 10 Pro  64-bit          10.0.19045 
```

</details>

**Comment by marcosmarxm** [comment]
Can you check the discussion on [Docker forum](https://forums.docker.com/t/tls-failed-to-verify-certificate-x509/137486). This looks your env is missing the certificates.

**Comment by atlasrule** [comment, marked as answer]
> Can you check the discussion on [Docker forum](https://forums.docker.com/t/tls-failed-to-verify-certificate-x509/137486). This looks your env is missing the certificates.

Thank you very much, solved the issue by installing on some different system and network.

**Comment by marcosmarxm** [comment]
@atlasrule can you provide more information about what system/network you tried and the one it worked?

**Comment by suhl79** [comment]
How come the OP marked this as an answer, when he actually had to reinstall airbyte on a different system to make it work. I'm experiencing exactly the same issue.

**Comment by dagleaves** [comment]
I was able to resolve the issue, at least temporarily so far. The issue was certificates expiring

From inside the airbyte-abctl-control-plane docker container:

```bash
kubeadm certs check-expiration
```

All had expired.

```bash
# Backup existing certificates
cp -r /etc/kubernetes/pki /etc/kubernetes/pki.backup

# Renew all certificates
kubeadm certs renew all

# Restart kubelet to pick up new certificates
systemctl restart kubelet

# Restart the container
exit
docker restart airbyte-abctl-control-plane
```

I was unable to update airbyte for potentially other reasons, but this was able to get the existing installation back up. I uninstalled and reinstalled (which keeps the data) and it came up fine.

**Comment by trahouston** [comment]
@dagleaves solution worked to get the connections running again
